### PR TITLE
dynstream: support area batch config and count+bytes batching

### LINKED ISSUE #4675: dynamic stream support batch by bytes
as title

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #4675 

This is the 2rd PR to support dynamic stream batch events by bytes at the area level.

dynstream currently only uses a shared batch config and does not have a common `count + bytes` batching core.

### What is changed and how it works?

- add area-level batch config support in dynstream
- add the core batcher for `count + bytes` batching
- make event queue and stream use the new batching logic
- add unit tests for batch config and batching behavior

#### Tests

- Unit test

##### Will it cause performance regression or break compatibility?

No obvious compatibility impact. This PR only changes dynstream internal batching logic.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note

```release-note
None
```

## Summary by CodeRabbit

* **New Features**
  * Per-area batch configuration with independent event-count and byte-size limits; streams and queues apply area-specific batching.

* **API Changes**
  * New area settings constructor to specify batch limits; handler callback return type and area-settings parameter names clarified.
  * Batch semantics: non-positive counts default to 1; negative byte limits treated as 0.

* **Tests**
  * Added and updated unit tests covering per-area batching, byte/count boundaries, first-add semantics, and edge cases.

### CODE REVIEW

File: utils/dynstream/area_batch_config_registry.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	cfg = newBatchConfig(cfg.softCount, cfg.hardBytes)
+	if cfg == s.defaultConfig {
+		return
+	}
+
+	s.setOverrideLocked(area, cfg)
+}
+
+// onRemovePath is not thread safe.
+func (s *areaBatchConfigRegistry[A]) onRemovePath(area A) {
```
Comment (wk989898) [inline utils/dynstream/area_batch_config_registry.go]: ditto

Comment (3AceShowHand) [reply utils/dynstream/area_batch_config_registry.go]: the lock is called by the outside caller.
